internal/pkg/build/build.go

```diff
@@ -251,19 +249,26 @@ func engineRequired(def types.Definition) bool {
 
 // runBuildEngine creates an imgbuild engine and creates a container out of our bundle in order to execute %post %setup scripts in the bundle
 func runBuildEngine(b *types.Bundle) error {
-	if syscall.Getuid() != 0 && !b.Opts.Fakeroot {
+	isPrivileged := syscall.Getuid() == 0
+
+	if !isPrivileged && !b.Opts.Fakeroot {
 		return fmt.Errorf("Attempted to build with scripts as non-root user or without --fakeroot")
 	}
 
 	sylog.Debugf("Starting build engine")
 	env := []string{sylog.GetEnvVar()}
-	starter := filepath.Join(buildcfg.LIBEXECDIR, "/singularity/bin/starter")
-	if b.Opts.Fakeroot {
-		starter = filepath.Join(buildcfg.LIBEXECDIR, "/singularity/bin/starter-suid")
-		if _, err := os.Stat(starter); os.IsNotExist(err) {
-			return fmt.Errorf("fakeroot feature requires to install Singularity as root")
-		}
+
+	requireSuid := b.Opts.Fakeroot && !isPrivileged
```

Comment:
deja vu.

Isn't the logic expressed on lines 252-256,261 repeated in a couple of places?